crates/stackable-operator/src/patchinator/mod.rs

```diff
@@ -0,0 +1,500 @@
+        source: kube::core::dynamic::ParseDynamicObjectError,
+        target_api_version: String,
+        target_kind: String,
+    },
+}
+
+pub fn apply_patches<R>(base: &mut R, patches: &ObjectOverrides) -> Result<(), Error>
+where
+    R: kube::Resource<DynamicType = ()> + DeepMerge + DeserializeOwned,
+{
+    for patch in &patches.object_overrides {
+        apply_patch(base, patch)?;
+    }
+    Ok(())
+}
+
+pub fn apply_patch<R>(base: &mut R, patch: &DynamicObject) -> Result<(), Error>
+where
+    R: kube::Resource<DynamicType = ()> + DeepMerge + DeserializeOwned,
+{
+    use kube::ResourceExt;
+
+    let Some(patch_type) = &patch.types else {
+        return Ok(());
+    };
+    if patch_type.api_version != R::api_version(&()) || patch_type.kind != R::kind(&()) {
+        return Ok(());
+    }
+    let Some(patch_name) = &patch.metadata.name else {
+        return Ok(());
+    };
+
+    // The name always needs to match
+    if &base.name_any() != patch_name {
+        return Ok(());
+    }
+
+    // If there is a namespace on the base object, it needs to match as well
+    // Note that it is not set for cluster-scoped objects.
+    if base.namespace() != patch.metadata.namespace {
+        return Ok(());
+    }
+
+    let deserialized_patch =
+        patch
+            .clone()
+            .try_parse()
+            .with_context(|_| ParseDynamicObjectSnafu {
+                target_api_version: R::api_version(&()),
+                target_kind: R::kind(&()),
+            })?;
+    base.merge_from(deserialized_patch);
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::BTreeMap, vec};
+
+    use k8s_openapi::{
+        ByteString, Metadata,
+        api::{
+            apps::v1::{
+                RollingUpdateStatefulSetStrategy, StatefulSet, StatefulSetSpec,
+                StatefulSetUpdateStrategy,
+            },
+            core::v1::{
+                ConfigMap, Container, ContainerPort, PodSpec, PodTemplateSpec, Secret,
+                ServiceAccount,
+            },
+            storage::v1::StorageClass,
+        },
+        apimachinery::pkg::util::intstr::IntOrString,
+    };
+    use kube::api::ObjectMeta;
+
+    use super::*;
+
+    /// Using [`serde_yaml`] to generate the test data
+    fn generate_service_account() -> ServiceAccount {
+        serde_yaml::from_str(
+            "
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: trino-serviceaccount
+  namespace: default
+  labels:
+    app.kubernetes.io/instance: trino
+    app.kubernetes.io/managed-by: trino.stackable.tech_trinocluster
+    app.kubernetes.io/name: trino
+  ownerReferences:
+  - apiVersion: trino.stackable.tech/v1alpha1
+    controller: true
+    kind: TrinoCluster
+    name: trino
+    uid: c85bfb53-a28e-4782-baaf-3c218a25f192
+",
+        )
+        .unwrap()
+    }
+
+    /// Generate the test data programmatically (as operators would normally do)
+    fn generate_stateful_set() -> StatefulSet {
```

Comment:
question: But, do we need to? I feel like this can also be a static YAML file (to actually only test the merging/overrides).